### DISCUSSION TITLE
Using readRect with WaveShare 1.28 inch LCD

### DISCUSSION THREAD

**Discussion by swrocks253** [Bodmer/TFT_eSPI#2856] [Compatible displays and setup files]
Does anyone know if the WaveShare LCD supports reading graphics using the readRect function in TFT_eSPI? I am running the AnimatedDial sample program on an ESP32 DevKit V1. Everything works until the dial moves. Instead of filling in the area of the old dial position with the correct background it is filling it with all white pixels. Everything else seems to work OK.

**Comment by swrocks253** [comment]
It looks like trt_Read_8() function is not working. This function is in TFT_eSPI.cpp around line 1394 in my version (2.5.0). It is always returning 255.
Code from TFT_eSPI:
`      // Read the 3 RGB bytes, colour is actually only in the top 6 bits of each byte
      // as the TFT stores colours as 18 bits
      uint8_t r = tft_Read_8();
      uint8_t g = tft_Read_8();
      uint8_t b = tft_Read_8();
      color = color565(r, g, b);`

**Reply by swrocks253** [reply]
Sorry, the function is tft_Read_8

**Comment by Bodmer** [comment]
That display has a DIM (data in) pin. It does not support reading from the CGRAM.

You can use sprites instead, see example here:
https://github.com/Bodmer/TFT_eSPI/discussions/860#discussioncomment-208303
